internal/error/errors.go

```diff
@@ -0,0 +1,84 @@
+package error
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	installConflictErrorPattern        = `Unable to continue with install: (\w+) "(.*)" in namespace "(.*)" exists and cannot be imported into the current release.*`
+	installConflictErrorGeneralPattern = `Unable to continue with install: .*`
+)
+
+// InstallConflictErr indicates one or more resources already exist in the cluster
+type InstallConflictErr error
+
+// UnknownErr indicates an error that cannot be classified
+type UnknownErr error
+
+func AsOlmErr(err error) error {
+	if err == nil {
+		return nil
+	}
+	for _, rule := range rules {
+		if matches := rule.MatchError(err); len(matches) > 0 {
+			// first index is the full match
+			logrus.Debugf("found match with params %v\n", matches)
+			return rule.TranslateError(err, matches[1:]...)
+		}
+	}
+
+	// let's mark any unmatched errors as unknown
+	return UnknownErr(err)
+}
+
+// errorTranslator is a function that translates an error into a more specific error
+// typically to hide implementation details
+type errorTranslator func(error, ...string) error
+
+// errorMatcher is a function that matches an error to a specific pattern and extracts any
+// relevant information - if nil is returned, no match was found
+type errorMatcher func(error) []string
+
+// rule links a matcher to a translator
+type rule struct {
+	matchFunc  errorMatcher
+	translator errorTranslator
+}
+
+func (r rule) MatchError(err error) []string {
+	return r.matchFunc(err)
+}
+
+func (r rule) TranslateError(err error, params ...string) error {
+	return r.translator(err, params...)
+}
+
+// rules is a list of rules for error translation
+var rules = []rule{
+	{regexErrorMatcher(installConflictErrorPattern), installConflictErrorTranslator},
+	{regexErrorMatcher(installConflictErrorGeneralPattern), installConflictErrorTranslator},
+}
```

Comment:
Should these use different functions? That might make `installConflictErrorTranslater()` below simpler... no need for a default `msg`, and no need to check for `len(params)`, which might be error prone.